### DISCUSSION TITLE
How to control if useFetch sends request on a page loading

### DISCUSSION THREAD

**Discussion by Pawel2505** [nuxt/nuxt#17335] [Questions] [answered]
I cannot determine how it works but it seems to be very strange. Lets suppose I have two pages A and B. The page A uses useFetch in setup. I observed that the useFetch works fine if I open page A very first time. Then when I navigate to B and back to page A useFetch does not work. Also when I reload page A it does not work! But iIf I reload page B and go back to page A, the useFetch works fine. 

I am totally confused. Sometimes I'd like to run request on every page loading, sometimes maybe only once, but how can I control it? I didn't find anything in docs about it.

**Comment by Flanker72** [comment, marked as answer]
`useFetch` is a rich wrapper for `$fetch` with SSR support. It cahses the request so it isn't refetched after the same page return.
You need to somehow trigger the data update (use `refresh` method exported by `useFetch` or `watch` argument).
Also consider that `useFetch` may cache its data during the SSR and pass it to the client-side, causing no additional request sent.
If you need to load data once, use `useFetch` in the `script setup`.
If you need to refresh data after every page reload, use Nuxt or Vue hooks like `onMounted` to trigger `refresh` manually.
